tests/framework/microvm.py

```diff
@@ -632,6 +632,12 @@ def spawn(
             )
             self._screen_pid = screen_pid
 
+        # If `--new-pid-ns` is used, the Firecracker process will detach from
+        # the screen and the screen process will exit. We do not want to
+        # attempt to kill it in that case to avoid a race condition.
```

Comment:
In that case we just shouldn't use screen. Which test is doing this?

We may also want to raise `if screen and new-pid-ns`

Reply:
In the microvm class. That combination of options cannot be handled by screen, so we should raise an exception when we detect it. I just don't think it's a valid use case.

Reply:
My point is we still use screen to start FC in this combination in `Microvm::spawn`, even though the screen itself exits soon after, so screen is still running for some time and we can't reliable assert its absence.
In order to get rid of screen usage for this use case we need to create a new helper that would allow to run a FC command that does not complete immediately (unlike in screen-detached or daemonised cases), which I'm not sure is warranted just by this case.
What kind of check do you mean? In the `Microvm::kill`?

Reply:
Synced offline. Doing this with the current codebase would be very complicated, so we will defer this until we extract the screen part as a separate "Jailer"